### ISSUE TITLE
coredns -plugins differ from plugin.cfg

### ISSUE THREAD

**Issue by cattyhouse** [coredns/coredns#6058]
**What happened**:

- `plugin.cfg` and build: 

```sh
# plugin.cfg

bufsize:bufsize
bind:bind
errors:errors
log:log
cache:cache
template:template
hosts:hosts
forward:forward

# build
go generate && go build

```
- `./coredns -plugins`

```
Server types:
  dns

Caddyfile loaders:
  flag
  default

Other plugins:
  dns.bind
  dns.bufsize
  dns.cache
  dns.debug
  dns.dnstap
  dns.errors
  dns.forward
  dns.hosts
  dns.log
  dns.metadata
  dns.prometheus
  dns.template
```

- 4 differs `coredns -plugins` vs `plugin.cfg`:

```
dns.debug
dns.dnstap
dns.metadata
dns.prometheus
```

**What you expected to happen**:

`coredns -plugins` should output same plugins as configured in `plugin.cfg`

**How to reproduce it (as minimally and precisely as possible)**:

see above steps

**Anything else we need to know?**:

though there are 4 differs in `coredns -plugins`, but they are not usable in Corefile, e.g. put `debug` or `prometheus localhost:9253` in Corefile results coredns : `Error during parsing: Unknown directive 'debug'`, `Error during parsing: Unknown directive 'prometheus'`

**Environment**:

- the version of CoreDNS: `v1.10.1`
- Corefile:
- logs, if applicable:
- OS (e.g: `cat /etc/os-release`): `Debian 12`
- Others:

**Comment by cattyhouse** [comment]
did some test, only enable `forward` and `cache` in `plugin.cfg`, these will show in `coredns -plugins`:

```
Other plugins:
  dns.cache
  dns.debug
  dns.dnstap
  dns.forward
  dns.metadata
  dns.prometheus
```

but these extra 4 plugins are not actually usable, the output may confuse users if they want to know **what plugins the binary coredns is built with** before setting up the `Corefile`.

**Comment by chrisohaver** [comment]
The output is confusing. I think this is due to cross plugin package imports (eg forward imports metadata). Coredns is actually built with those plugin packages. However, if they aren’t explicitly in the plugins.cfg, they are not usable in the Corefile. 

I think this could be “fixed” by refactoring all plugin-plugin dependencies, moving shared packages to plugin/pkg. This would be a fairly tedious task though.

Another way would be to amend the `-plugins` code to differentiate usable plugins from in memory only plugins.

**Comment by cattyhouse** [comment]
yeah, option 2 would be easier, a mark like : `dns.prometheus(import only)` would be enough for users to distinguish.
